### PR TITLE
Reduced minimum spring version requirement to 4.1.6

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Description

(does not apply to spring boot starter and auto configuration)

## Motivation and Context

Allow usage of core Riptide modules in older Spring versions.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-chaos/src/main/java/org/zalando/riptide/chaos/EmptyInputStream.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.zalando.riptide.chaos;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+final class EmptyInputStream extends InputStream {
```
Comment (lukasniemeier-zalando) [inline riptide-chaos/src/main/java/org/zalando/riptide/chaos/EmptyInputStream.java]: this incarnation of this class is untested :stuck_out_tongue_closed_eyes:

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: riptide-chaos/src/main/java/org/zalando/riptide/chaos/EmptyInputStream.java (hunk 1 above)
Comment (whiskeysierra) [reply riptide-chaos/src/main/java/org/zalando/riptide/chaos/EmptyInputStream.java]: How do you know?

Comment (lukasniemeier-zalando) [reply riptide-chaos/src/main/java/org/zalando/riptide/chaos/EmptyInputStream.java]: I only see a test for the one of `riptide-httpclient`... but I also see the similarities between the implementations.

**Comment by whiskeysierra** [comment]
It's covered implicitly by other unit tests

**Pull request merged**